gemrb/core/Variables.h

```diff
@@ -109,8 +109,27 @@ class GEM_EXPORT Variables {
 
 	// Operations
 	void SetAtCopy(const char* key, const char* newValue);
-	void SetAtCopy(const char* key, int newValue);
-	void SetAt(const char* key, const String& newValue);
+	
+	template<typename STR, ENABLE_CHAR_RANGE>
+	void SetAtCopy(const STR& key, int newValue)
+	{
+		const std::string& newStr = fmt::format("{}", newValue);
+		SetAtCopy(&key[0], newStr.c_str());
+	}
+	
+	template<typename STR, ENABLE_CHAR_RANGE>
+	void SetAt(const STR& key, const String& newValue)
+	{
+		const std::string& mbstr = MBStringFromString(newValue);
+		SetAtCopy(&key[0], mbstr.c_str());
+	}
+	
+	template<typename STR, ENABLE_CHAR_RANGE>
+	void SetAt(const STR& key, const std::string& newValue)
```

Comment:
Should these two really be called SetAt, since they call SetAtCopy? We're hazing the distinction

Reply:
it wont matter soon, AFIK the only distinction is that the copy variant `strdup`s the string. since there are so few uses of that we should just make all public methods the same.

Reply:
oh, I guess it returns the default, which in this case is '0'

Reply:
Yes, it's mostly used for tlk tokens, but that's not so rare. Did you add another Variables type for std::string to make the strdups redundant?

Reply:
testing by making `void SetAt(const char* key, const char* newValue);` private showed that there are no remaining callers for that flavor.